pkg/gpu/cuda/fatbin.go

```diff
@@ -0,0 +1,475 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:generate go run golang.org/x/tools/cmd/stringer@latest -output fatbin_string.go -type=nvInfoAttr,nvInfoFormat,fatbinDataKind -linecomment
+
+// Package cuda provides helpers for CUDA binary parsing
+package cuda
+
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"unsafe"
+
+	"github.com/pierrec/lz4/v4"
+)
+
+type fatbinDataKind uint16
+
+const fatbinDataKindPtx fatbinDataKind = 1
+const fatbinDataKindSm fatbinDataKind = 2
+
+const fatbinMagic = 0xBA55ED50
```

Comment:
I hadn't seen it before. Looking usages of the same number it looks like it might be the format for CUDA 3.2 and earlier, which was released in 2010. I don't think it's needed to support it.